src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -0,0 +1,148 @@
+// @flow
+
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { injectIntl } from 'react-intl';
+import MultiGrid from 'react-virtualized/dist/es/MultiGrid/MultiGrid';
+import AutoSizer from 'react-virtualized/dist/es/AutoSizer';
+import FileIcon from '../../icons/file-icon';
+import messages from '../../elements/common/messages';
+import { getFileExtension } from '../../utils/file';
+import '../../elements/content-explorer/ItemList.scss';
+import './MetadataBasedItemList.scss';
+
+const FILEICON_SIZE = 32;
+const FILEICON_COLUMN_WIDTH = 52;
+const FILENAME_COLUMN_WIDTH = 350;
+const MIN_METADATA_COLUMN_WIDTH = 250;
+const FIXED_COLUMNS_NUMBER = 2; // 2 Sticky columns - 1. file-icon, 2. file-name for each row
+const FIXED_ROW_NUMBER = 1; // Header row
+
+type Props = {
+    currentCollection: Object,
+    intl: Object,
+    metadataColumnsToShow: Array<string>,
+} & InjectIntlProvidedProps;
+
+type CellRendererArgs = {
+    columnIndex: number,
+    key: string,
+    rowIndex: number,
+    style: Object,
+};
+
+class MetadataBasedItemList extends Component<Props> {
+    props: Props;
+
+    getColumnWidth(width) {
+        const { metadataColumnsToShow }: Props = this.props;
+        const availableWidth = width - FILENAME_COLUMN_WIDTH - FILEICON_COLUMN_WIDTH; // total width minus width of sticky columns
```

Comment:
Should this have an order to the operations?

Reply:
It won't matter. Evaluated from ltr.